android/app/src/main/java/org/smartregister/fhircore/viewmodel/PatientListViewModel.kt

```diff
@@ -129,6 +131,25 @@ class PatientListViewModel(application: Application, private val fhirEngine: Fhi
     viewModelScope.launch { fhirEngine.syncUpload() }
   }
 
+  fun isPatientExists(id: String, callback: OnPatientSearchResult) {
```

Comment:
The `View Model` should not hold a reference to the [View](https://developer.android.com/topic/libraries/architecture/viewmodel). This is a potential source of memory leaks. Instead of passing a Call back from the view, observe the results of the ViewModel and pass a variable with conditional values in a `LiveData` object i.e. Handle the results on the view

````
  fun findPatient(id: String) : LiveData<Result<Patient>> {
    val result = MutableLiveData<Result<Patient>>()
    viewModelScope.launch {
      try {
        val patient = fhirEngine.load(Patient::class.java, id)
        result.value = Result.success(patient)
      } catch (e: ResourceNotFoundException) {
        result.value = Result.failure(e)
      }
    }
    return result
  }

```

Reply:
Done